File: C_Programming/Practise_Programs/PP_143.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_143
//
//  Description       : This program demonstrates the bitwise AND,
//                      OR and XOR operations on two integer numbers.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 17/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int main()
{
    int iNo1 = 0, iNo2 = 0, iAns = 0 ;

    printf("Enter First Number : \n");
    scanf("%d", &iNo1);

    printf("Enter Second Number : \n");
    scanf("%d", &iNo2);
    
    iAns = iNo1 & iNo2;

    printf("AND : %d\n", iAns);

    iAns = iNo1 | iNo2;

    printf("OR : %d\n", iAns);

    iAns = iNo1 ^ iNo2;

    printf("XOR : %d\n", iAns);
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      12
//      10
//
//  Sample Output :
//
//      Enter First Number :
//      Enter Second Number :
//      AND : 8
//      OR : 14
//      XOR : 6
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_143.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_143
//
//  Description       : This program prints a hollow rectangle
//                      pattern using nested for loops. The first
//                      and last rows are completely filled with
//                      stars, while the middle rows contain stars
//                      only at the beginning and end.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 31/08/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

class PP_143
{
    public static void main(String args[])
    {
        for(int i = 1; i <= 1; i++)
        {
            for(int j = 1; j <= 5; j++)
            {
                System.out.print("*");
            }

            System.out.println();
        }

        for(int i = 1; i <= 4; i++)
        {
            System.out.print("*");

            for(int j = 1; j <= 3; j++)
            {
                System.out.print(" ");
            }

            System.out.print("*");

            System.out.println();
        }

        for(int i = 1; i <= 1; i++)
        {
            for(int j = 1; j <= 5; j++)
            {
                System.out.print("*");
            }
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      *****
//      *   *
//      *   *
//      *   *
//      *   *
//      *****
//
//////////////////////////////////////////////////////////////////